merkle/compact/tree.go

```diff
@@ -256,9 +257,9 @@ func (t *Tree) AddLeafHash(leafHash []byte, setNodeFn setNodeFunc) (int64, error
 			return assignedSeq, nil
 		}
 		// The bit is set so we have a node at that position in the nodes list so hash it with our running hash:
-		hash = t.hasher.HashChildren(t.nodes[bit], hash)
-		// Store the resulting parent hash.
-		if err := setNodeFn(bit+1, index, hash); err != nil {
+		hash := t.hasher.HashChildrenInto(t.nodes[bit], hash, hash)
```

Comment:
It seems that physically `:=` doesn't change the algorithm (only luckily, because the `hash` size doesn't change), but logically it makes less sense. This line depends on the outer `hash` left from the previous iteration, but it doesn't update it for the next iteration explicitly. I suggest leaving `=`.

Reply:
That was unintended. I'll fix it.